arcgis-runtime-samples-macos/Content Display Logic/Controllers/SampleListViewController.swift

```diff
@@ -178,7 +178,7 @@ extension SampleListViewController: NSOutlineViewDataSource {
         case let sampleItem as SampleItem:
             return sampleItem.value.name
         default:
-            fatalError()
```

Comment:
Which rule prompted this change? `fatalError()` is appropriate here because it represents an impossible condition. `item` will either be a `CategoryItem` or a `SampleItem`, but there is no way for the compiler to know that. Returning `nil` makes it seem like `default` is an expected condition.

Reply:
`fatal_error_message` flagged the call. I didn't see any harm in replacing it instead of writing a message since the code will never be run anyway.

Reply:
I would prefer either not enabling that rule or modify this usage of `fatalError()` to include a message, something like this:

```suggestion
            fatalError("item should only ever be a CategoryItem or SampleItem")
```

Reply:
Sure. Done.